sync.md

```diff
@@ -181,29 +164,61 @@ func (syncer *Syncer) collectChainCaughtUp(maybeHead TipSet) (Chain, error) {
+
+After collecting a chain up to an ancestor TipSet that was previously synced to the store the syncing protocol checks each TipSet of the new chain for validity one by one.  When the filecoin network runs Expected Consensus, or any other multiple parents consensus protocol, the syncing protocol must consider not only the TipSets in the new chain but also possible new-heaviest TipSets that are the union of TipSets in the new chain and TipSets already in the store.  In the case of Expected Consensus there is at most one such TipSet: the TipSet made up of the union of the first new TipSet in the new chain being synced and the largest TipSet with the same parents kept in the store.
 
-To sync new tipsets the `caught up` syncing protocol first runs a consensus validation check on the tipset.  If any tipset is invalid the syncing protocol finishes.  If a tipset is valid the syncer adds the tipset to the chain store.  The syncing protocol then checks whether the tipset is heavier than the current head using the consensus weighting rules.  If it is heavier the chain updates the state of the node to account for the new heaviest tipset.
+To sync new TipSets the `caught up` syncing protocol first runs a consensus validation check on the TipSet.  If any TipSet is invalid the syncing protocol finishes.  If a TipSet is valid the syncer adds the TipSet to the chain store.  The syncing protocol then checks whether the TipSet is heavier than the current head using the consensus weighting rules.  If it is heavier the chain updates the state of the node to account for the new heaviest TipSet.
+
+## Maintaining a fresh Peer Set
+
+Syncing depends on the validity of a node's peer set. In order to ensure that the peer set remains representative of the network's state after bootstrap, a node should regularly update its peer set.
+
+```go
+func (syncer *Syncer) updatePeerSet() {
+    for peer, info := range syncer.PeerSet {
+        if !trustedPeer && syncer.TimeNow() - info.initialConnection > PEER_RECYCLE {
+            syncer.replacePeer(peer)
+        }
+        
+        if syncer.TimeNow() - peer.lastConnection > PEER_HEARTBEAT {
+            response, err := Heartbeat(peer)
+            if err {
+                syncer.replacePeer(peer)
+            } else {
+                syncer.PeerHeads(response.HeaviestTipSet)
+            }
+        }
+    }
+}
+
+func (syncer *Syncer) replacePeer(peer PeerID) {
+    delete(syncer.PeerSet, peer)
+    delete(syncer.PeerHeads, peer)
+            
+    syncer.getNewPeer()
```

Comment:
We should define the requirements here. What should this method do?

Reply:
My point here was to keep the peer set chaning through time. Cycle out nodes on an untrusted basis and replace peers as you go to keep making sure you're tracking the appropriate chain head. 

This could be overkill to start. Without a well defined heartbeat, it may be worth just ripping out altogether. Kill or keep?